### PR TITLE
Fix: 1745 milestone values persist

### PR DESCRIPTION
**Pull request by Sepehr-Sobhani**
[ZH CARD 1745](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1745)

### CODE REVIEW

**Review by BCerki — COMMENTED** [review]
Nice work!

A couple things:
1) This PR removes the calculated values from the payments table, but I wonder if it should remove the payment record entirely? 
2) I'm seeing a funky diff for holdback: 
image
This happened when I created a new project, created a General Milestone, and filled out everything. Then in my revision I changed the type to a Reporting Milestone

File: app/components/Form/Functions/getMilestoneFilteredSchema.ts [hunk 1]
```diff
@@ -35,9 +35,14 @@ export const getMilestoneFilteredSchema = (
   const newDataObject: any = {};
 
   for (const key of Object.keys(filteredSchema.properties)) {
+    const [updatedFormData, prevFormData] = [
```
Comment (BCerki) [inline app/components/Form/Functions/getMilestoneFilteredSchema.ts:38]: Nice, two bug fixes in one! No suggested change here, but FYI, there's a similar thing going on in the `schemaFilteringHelpers`

**Review by BCerki — COMMENTED** [review]
Nice work!

I found another redundant arrow that I added to your bug card: https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1822

File: schema/deploy/functions/handle_milestone_form_change_commit.sql [hunk 2]
```diff
@@ -89,12 +99,32 @@ begin
...
-      net_amount = coalesce((fc.new_form_data->>'adjustedNetAmount')::numeric, (fc.new_form_data->>'calculatedNetAmount')::numeric),
-      date_sent_to_csnr = (fc.new_form_data->>'dateSentToCsnr')::timestamptz
-    where py.reporting_requirement_id = fc.form_data_record_id;
+    if report_is_eligible_for_expenses then
+      -- This part is covering the case where the user changes the report type from a non-expense report type to an expense report type (or vice versa)
+      -- if there's a payment record with the same reporting_requirement_id, we update it
+      if exists(select 1 from cif.payment where reporting_requirement_id = fc.form_data_record_id and archived_at is null) then
+        update cif.payment set
+          gross_amount = adjusted_or_calculated_gross_amount,
+          net_amount = adjusted_or_calculated_net_amount,
```
[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply schema/deploy/functions/handle_milestone_form_change_commit.sql:108]: @BCerki I'm not quite sure if I understand, can we pair on this?

File: schema/test/unit/functions/handle_milestone_form_change_commit_test.sql [hunk 3]
```diff
@@ -224,6 +232,167 @@ select results_eq(
...
+  $$
+  values (
+  '2021-10-31 14:24:46.318423-07'::timestamptz,
+  '2021-09-30 14:24:46.318423-07'::timestamptz,
+  'reporting milestone comments'::varchar,
+  1
+  )
+  $$,
+  'The correct values were are preserved in the reporting_requirement table on update'
+);
```
[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply schema/test/unit/functions/handle_milestone_form_change_commit_test.sql:271]: wrong description :)

File: schema/test/unit/functions/handle_milestone_form_change_commit_test.sql [hunk 4]
```diff
@@ -224,6 +232,167 @@ select results_eq(
...
+  $$,
+  $$
+  values (
+  '2021-10-31 14:24:46.318423-07'::timestamptz,
+  '2021-09-30 14:24:46.318423-07'::timestamptz,
+  'advance new comments'::varchar,
+  1
+  )
+  $$,
+  'The correct values were are preserved in the reporting_requirement table on update'
```
[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply schema/test/unit/functions/handle_milestone_form_change_commit_test.sql:354]: wrong description :)

File: schema/test/unit/functions/handle_milestone_form_change_commit_test.sql [hunk 5]
```diff
@@ -182,9 +188,10 @@ select results_eq(
   1
   )
   $$,
-  'The correct values were added to the reporting_requiement table on update'
+  'The correct values were added to the reporting_requirement table on update'
```
[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply schema/test/unit/functions/handle_milestone_form_change_commit_test.sql:191]: I didn't change the description of this, I only fixed a wording issue but you are right, it makes more sense and I'm going to change it :)

**Review by BCerki — APPROVED** [review]

**Pull request merged**